### ISSUE TITLE
DTD build issue

### ISSUE THREAD

**Issue by martindholmes** [TEIC/Stylesheets#185]
The tei_all.dtd file appears to have an error:

https://www.tei-c.org/release/xml/tei/custom/schema/dtd/tei_all.dtd

Line 3633:

type (met|real|rhyme) "met real" 

Error: The defaultValue "met real" of attribute "type" is not legal as for the lexical constraints of this attribute type.

This is not a result of the dtdMagic.xsl issue that Hugh has recently solved. Don't know the cause yet...

**Comment by martindholmes** [comment]
I think the problem is caused by the fact that the attribute @type is defined as being teidata.enumerated, and the default value has two tokens in it (stripped-down view):

~~~~~
<datatype minOccurs="1" maxOccurs="3"><dataRef key="teidata.enumerated"/></datatype>
<defaultVal>met real</defaultVal>
<valList type="closed">
  <valItem ident="met">
    <gloss versionDate="2007-05-12" xml:lang="en"><att>met</att> attribute</gloss>
  </valItem>
  <valItem ident="real">
    <gloss versionDate="2007-05-12" xml:lang="en"><att>real</att> attribute</gloss>
  </valItem>
  <valItem ident="rhyme">
    <desc versionDate="2007-05-12" xml:lang="en">declaration applies to the rhyme scheme recorded on the <att>rhyme</att> attribute</desc>
  </valItem>
</valList>
~~~~~

**Comment by hcayless** [comment]
Just pushed a fix. Though probably should have run the tests first, as it
may affect some expected results...

**Comment by martindholmes** [comment]
It's the only instance of this datatype with a multiple token in a defaultVal. We should look carefully at this in the light of the war on defaultVals.

**Comment by lb42** [comment]
If you look back at the definition for this element in P4 (http://www.tei-c.org/Vault/GL/P3/ref/METDECL.htm) you'll see that the situation is a tad more complex. Essentially what's happening here is that we have specified teidata.enumerated with a maxOccurs which permits multiple valuest. But in DTDland you can only specify EITHER an alternation of specific values (one only permitted) OR one or more names (values not specifiable) -- which is what "NAMES" means in the DTD for this element. So we ought to be looking at the cardinality of our teidata.enumerated values when generating DTDs. Don't know how widespread or significant this problem is, but it's nowt to do with default values. That's just the canary which enabled us to detect the problem.

**Comment by hcayless** [comment]
Fixed with 7aa37cdb006d0f8b27cb775e3225597f6c4fbd2c
